### PR TITLE
refactor: kausal.libsonnet to match k8s-alpha

### PR DESCRIPTION
**Pull request by Duologic**
Essentially a refactoring of kausal.libsonnet to work with k8s-alpha. This might break compat with ksonnet-lib, therefore moved to a new path/file.

ref: #215

Might want to diff locally against the original file: `diff ksonnet-util/kausal.libsonnet k8s-util/k8s.libsonnet`

### CODE REVIEW

File: tanka-util/util.libsonnet [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+        withData(data)::
+          if (data == {}) then {}
+          else super.withData(data),
+        withDataMixin(data)::
+          if (data == {}) then {}
+          else super.withDataMixin(data),
+      },
+
+      // Expose containerPort type.
+      containerPort:: $.core.v1.containerPort {
```
Comment (sh0rez) [inline tanka-util/util.libsonnet:25]: No need to reference it, +: works

File: tanka-util/util.libsonnet [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+          super.newNamed(name=name, containerPort=port),
+
+        // Shortcut constructor for UDP ports.
+        newUDP(name, port)::
+          super.newNamed(name=name, containerPort=port) +
+          super.withProtocol('UDP'),
+      },
+
+      // Expose volumes type.
+      volume:: $.core.v1.volume {
```
Comment (sh0rez) [inline tanka-util/util.libsonnet:37]: Same as container

File: tanka-util/util.libsonnet [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+        // Make items parameter optional from fromConfigMap
+        fromConfigMap(name, configMapName, configMapItems={})::
+          super.fromConfigMap(name, configMapName, configMapItems),
+
+        // Rename emptyDir to claimName
+        fromPersistentVolumeClaim(name='', claimName='')::
+          super.fromPersistentVolumeClaim(name=name, emptyDir=claimName),
+      },
+
+      volumeMount:: $.core.v1.volumeMount {
```
Comment (sh0rez) [inline tanka-util/util.libsonnet:47]: Same here

File: tanka-util/util.libsonnet (hunk 1 above)
Comment (Duologic) [reply tanka-util/util.libsonnet]: Excellent catch, originally this pulled in from `container.portsType`.

**Comment by Duologic** [comment]
I've made a branch here to see what has changed: https://github.com/grafana/jsonnet-libs/compare/duologic/diff_branch (do not merge this)

**Comment by Duologic** [comment]
replaced by #293

**Pull request closed without merging**